sdk/src/main/java/io/dapr/client/domain/GetSecretRequest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+package io.dapr.client.domain;
+
+import io.grpc.Context;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A request to get a secret by key.
+ */
+public class GetSecretRequest {
+
+  private String secretStoreName;
+
+  private String key;
+
+  private Map<String, String> metadata = new HashMap<>();
+
+  private Context context;
+
+  public String getSecretStoreName() {
+    return secretStoreName;
+  }
+
+  void setSecretStoreName(String secretStoreName) {
+    this.secretStoreName = secretStoreName;
+  }
+
+  public String getKey() {
+    return key;
+  }
+
+  void setKey(String key) {
+    this.key = key;
+  }
+
+  public Map<String, String> getMetadata() {
+    return metadata;
+  }
+
+  void setMetadata(Map<String, String> metadata) {
+    this.metadata = metadata;

```

Comment:
unmodifiableMap? same for other requests?

Reply:
This method is not public on purpose. The builder classes take care of the unmodifiableMap.